### ISSUE TITLE
cv_run copies the output tensor into a 16-byte buffer with no clamp; any non-classifier model overflows the stack

### ISSUE THREAD

**Issue by victor-wildlife** [wildlifeai/Seeed_Grove_Vision_AI_Module_V2#225]
**What:** `cv_run` copies `g_class_count` bytes into a 16-byte stack buffer with no clamp, and `g_class_count` is derived from the output tensor's last dimension. Any non-classifier model overflows the stack; the class/label mismatch that would flag it is only a warning.

**Where:** `ww500_md/cvapp.cpp:609`, `:818`, `:835`; `ww500_md/image_task.c:768`; `ww500_md/xip_manager.h:50`

Read on `dev` at `3ca2823c`. Not observed on hardware; found by reading the path while tracing wildlifeai/ww-website#134.

**Evidence / context:**

The class count is derived in two places with two different formulas:

| | formula | used for |
|---|---|---|
| `cvapp.cpp:609` (model load) | `dims->data[dims->size - 1]`, the **last** dim | `g_class_count` |
| `cvapp.cpp:835` (`outputAsPercentage`) | `dims->data[1]`, the **second** dim | softmax width |

They agree only for a rank-2 `[1, C]` classifier. `g_class_count` is a `uint8_t`, so it also truncates.

`cv_run` then copies without a bound:

```c
// cvapp.cpp:816
* categoriesCount = g_class_count;
for (uint8_t i = 0; i < g_class_count; i++)  {
    outCategories[i] = output->data.int8[i];
}
```

into a buffer the caller declares as `int8_t outCategories[MAX_CLASSES]` (`image_task.c:768`), where `MAX_CLASSES` is 16 (`xip_manager.h:50`).

For the models in this repo's own `model_zoo`, all of which the website's pretrained catalogue offered for deployment until wildlifeai/ww-website#139:

| model | output shape | last dim | `g_class_count` (u8) | writes into `int8_t[16]` |
|---|---|---|---|---|
| person detect (`allon_sensor_tflm`) | `[1, 2]` | 2 | 2 | safe |
| `rat_detection` | `[1, 2]` | 2 | 2 | safe |
| `tflm_yolov8_pose` | `[1, 256, 64]` | 64 | 64 | **48 bytes past the end** |
| `tflm_yolo11_od` 224 | `[1, 28, 28, 144]` | 144 | 144 | **128 bytes past** |
| `tflm_yolov8_od` 192 | `[1, 4, 756]` | 756 | **244** | **228 bytes past** |
| `tflm_yolo11_od` 192 | `[1, 84, 756]` | 756 | **244** | **228 bytes past** |

Two more problems in `outputAsPercentage` on the same models:

- `metaDataFlash->labels[i]` is read for `i < num_classes` (84 for YOLOv11) out of `char labels[MAX_CLASSES][MAX_LABEL_LEN]`. The store into `g_last_confidence_data` is guarded by `i < MAX_CLASSES`, but the read is not, and because a 2D-array row is never NULL the `if (label)` guard always passes and `xprintf("%s", label)` dereferences it.
- `float logits_float[num_classes]` and `float probabilities[num_classes]` are stack VLAs: 2 KB for the pose model.

The check that would catch all of this exists and only warns:

```c
// cvapp.cpp:618
if (g_class_count != metaDataFlash->class_count) {
    xprintf("WARNING: Number of classes and labels unmatched\n");
}
```

This is the firmware half of wildlifeai/ww-website#134, where a two-class person detector shipped with a one-line labels file and the device reported class 1 as `''`. The website now refuses to package that (LM-1); the device should refuse to load it.

Downstream, `processNNOutput` (`image_task.c:2458`) hardcodes the target as `outCategories[1]` above its own `// TODO This only works for the person detection`. For a YOLO tensor that byte is a bounding-box coordinate channel, so "TARGET OBJECT DETECTED" would be driven by geometry.

**Suggested fix:**

1. **Clamp the copy** in `cv_run` to `MAX_CLASSES`, and make `g_class_count` large enough not to truncate, or clamp at assignment.
2. **One class-count formula.** `ww500_md` is a classifier pipeline: it softmaxes a vector and reports index 1 as the target. Accept only a rank-1 or rank-2 output, take `C` from it, and reject anything else at load with a clear message. That makes the classifier-only contract explicit rather than an accident of which dimension gets read.
3. **Refuse to load on mismatch** at `cvapp.cpp:618` instead of warning. The website side of the same check is wildlifeai/ww-website#139; the two should agree on the rule: every class has a label, in order, and `C <= MAX_CLASSES`.
4. Bound the `labels[i]` read (use `cv_getLabel`, which already checks), and the VLAs go away once `C <= MAX_CLASSES` is enforced.

Related, not duplicates: #189 decides logits vs percentages and #185 moves `USE_PERCENTAGE` out of `image_task.c`; neither covers the buffer size or the class-count derivation. The detection models are recorded as future work on the website side, gated on person and rat detection running end to end first.
